libevmasm/Assembly.cpp

```diff
@@ -1021,18 +1149,17 @@ LinkerObject const& Assembly::assemble() const
 			break;
 		case AssignImmutable:
 		{
-			assertThrow(!eof, AssemblyException, "AssignImmutable in EOF code");
 			// Expect 2 elements on stack (source, dest_base)
 			auto const& offsets = immutableReferencesBySub[i.data()].second;
-			for (size_t i = 0; i < offsets.size(); ++i)
+			for (size_t offsetIdx = 0; offsetIdx < offsets.size(); ++offsetIdx)
```

Comment:
Also, just a small clarification here regarding https://github.com/ethereum/solidity/pull/15367#discussion_r1763372443.

 TBH, `i` used for a loop counter is the one idiomatic case where I find a single char name perfectly acceptable :) It's the outer loop that uses `i` to refer to `AssemblyItem` that bothered me way more :) But no need to change it now. Just wanted to point this out.

Reply:
Yes I changed it because the main look uses `i` and my IDE complained about that. I decided to change this one because wanted to keep the idea to not change to much foe legacy. But now when I'm looking on it maybe it's worth changing `i` in outer loop.

Reply:
Fair enough :)

IMO it's fine not to touch that outer `i` here. Less noise in the diff :) I only pointed this out because it was the opposite of what I'd do so just wanted to make sure we're on the same page about this. But looks like we are.

Reply:
Fixed it and changed also in legacy.